apps/prettier-git/src/prettier-git.ts

```diff
@@ -0,0 +1,448 @@
+    let content: string = '';
+    const decoder: StringDecoder = new StringDecoder('utf-8');
+    for await (const data of catFileProcess.stdout) {
+      content += decoder.write(data);
+    }
+    content += decoder.end();
+
+    await once(catFileProcess, 'close');
+
+    yield {
+      filePath,
+      metadata,
+      content
+    };
+  }
+}
+
+function showDiff(filePath: string, formatted: string): number {
+  // Checkout the files in the index to the working tree
+  const contextLines: number = 3;
+  const diffProcess: child_process.SpawnSyncReturns<string> = child_process.spawnSync(
+    'git',
+    [
+      '--no-optional-locks',
+      'diff',
+      '--no-index',
+      '--minimal',
+      '--color=always',
+      `--src-prefix=original/`,
+      `--dst-prefix=prettier/`,
+      `--unified=${contextLines}`,
+      '--',
+      filePath,
+      '-'
+    ],
+    {
+      input: formatted,
+      encoding: 'utf-8'
+    }
+  );
+  const output: string = diffProcess.stdout.toString();
+  const diffLine: string[] = /@@\s+[-](\d+),(\d+)/.exec(output)!;
+  let affectedLine: number = Number(diffLine[1]);
+  const lineCount: number = Number(diffLine[2]);
+
+  // Compensate for the context (imperfect at start of file)
+  affectedLine += Math.min(contextLines, lineCount - contextLines - 1);
+
+  console.log(output);
+  return affectedLine;
+}
+
+(async () => {
+  interface IPrettierIgnorer {
+    ignores(filePath: string): boolean;
+  }
+
+  interface IPrettierInternal {
+    __internal: {
+      createIgnorer(ignorePath: string, withNodeModules: boolean): IPrettierIgnorer;
+    };
+  }
+  const ignorer: IPrettierIgnorer = await (prettier as unknown as IPrettierInternal).__internal.createIgnorer(
+    path.resolve(process.cwd(), '.prettierignore'),
+    false
+  );
+  const shouldIgnoreFile: (filePath: string) => boolean = (filePath) => ignorer.ignores(filePath);
+
+  const fileMap: Map<string, IFileInfo> = await readIndexDiff(true, shouldIgnoreFile, isInQuietMode);
+  if (syncWorkingTree) {
+    // If asked to sync the working tree, we need to know if files are fully staged or not
+    // We do this by comparing the staged-only diff to the working tree diff
+    const partialMeta: Map<string, IFileInfo> = await readIndexDiff(false, shouldIgnoreFile, true);
+    for (const [filePath, item] of partialMeta) {
+      const entry: IFileInfo | undefined = fileMap.get(filePath);
+      if (entry && !item.staged) {
+        entry.fullyStaged = false;
+      }
+    }
+  }
+
+  const indexInfoLines: string[] = [];
+  const filesToClean: string[] = [];
+  let checkFailed: boolean = false;
+  let hasErrors: boolean = false;
+
+  const configPath: string = path.resolve(process.cwd(), '.prettierrc.js');
+
+  for await (const file of readFileContents(fileMap)) {
+    const { filePath, metadata, content } = file;
+
+    try {
+      const prettierConfig: prettier.Options | null = await prettier.resolveConfig(filePath, {
+        config: configPath
```

Comment:
Removing this option allows resolving from subtrees (consider replacing with `cache: true`) but that doesn't play nicely with checked-in bundled dependencies, since nested configs can add references to new plugins.